engines/stark/stark.cpp

```diff
@@ -185,24 +185,24 @@ void StarkEngine::processEvents() {
 
 		if (isPaused()) {
 			// Only pressing key P to resume the game is allowed when the game is paused
-			if (e.type == Common::EVENT_KEYDOWN && e.kbd.keycode == Common::KEYCODE_p) {
+			if (e.type == Common::EVENT_CUSTOM_ENGINE_ACTION_START && e.customType == kActionPause) {
 				_gamePauseToken.clear();
 			}
 			continue;
 		}
 
-		if (e.type == Common::EVENT_KEYDOWN) {
+		if (e.type == Common::EVENT_KEYDOWN || e.type == Common::EVENT_CUSTOM_ENGINE_ACTION_START) {
 			if (e.kbdRepeat) {
 				continue;
 			}
 
-			if (e.kbd.keycode == Common::KEYCODE_p) {
+			if (e.kbd.keycode == Common::KEYCODE_p || e.customType == kActionPause) {
 				if (StarkUserInterface->isInGameScreen()) {
 					_gamePauseToken = pauseEngine();
 					debug("The game is paused");
 				}
 			} else {
-				StarkUserInterface->handleKeyPress(e.kbd);
+				StarkUserInterface->handleKeyPress(e.kbd, e.customType);
```

Comment:
One thing I would like to check: When this event is EVENT_KEYDOWN, are we certain customType is zero (equal to kActionNone)?

Reply:
You can ignore my prior statement. I am now confident customType and kbd are both correctly set to correct defaults when the event type does not pertain to them.